kraken-build/src/kraken/std/python/tasks/ruff_task.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+from dataclasses import dataclass
+from pathlib import Path
+
+from kraken.common import Supplier
+from kraken.core import Project, Property
+from kraken.std.python.tasks.pex_build_task import pex_build
+
+from .base_task import EnvironmentAwareDispatchTask
+
+
+class RuffTask(EnvironmentAwareDispatchTask):
+    """A task to run `ruff` in either format, fix, or check mode."""
+
+    description = "Lint Python source files with Ruff."
+    python_dependencies = ["ruff"]
+
+    ruff_bin: Property[str] = Property.default("ruff")
+    ruff_task: Property[str] = Property.default("")
+    config_file: Property[Path]
+    additional_args: Property[Sequence[str]] = Property.default_factory(list)
+
+    def get_execute_command(self) -> list[str]:
+        command = [
+            self.ruff_bin.get(),
+            self.ruff_task.get(),
+            str(self.settings.source_directory),
+            *self.settings.get_tests_directory_as_args(),
+        ]
+        command += [str(directory) for directory in self.settings.lint_enforced_directories]
+        if self.config_file.is_filled():
+            command += ["--config", str(self.config_file.get().absolute())]
+        command += self.additional_args.get()
+        return command
+
+
+@dataclass
+class RuffTasks:
+    check: RuffTask
+    fix: RuffTask
+    format: RuffTask
```

Comment:
Do we want also a 4th task to check that the code formatting is valid (`format --check`) just like what is done for Black?

Reply:
This is the point bring in upper comment. I'll add the fourth rule for checking formatting is valid. 👍